rust/src/web/graphql/feetransfers/mod.rs

```diff
@@ -0,0 +1,237 @@
+    #[graphql(name = "BLOCKHEIGHT_ASC")]
+    BlockHeightAsc,
+    #[graphql(name = "BLOCKHEIGHT_DESC")]
+    BlockHeightDesc,
+}
+
+#[derive(Default)]
+pub struct FeetransferQueryRoot;
+
+/// convert epoch millis to an ISO 8601 formatted date
+fn millis_to_date_string(millis: i64) -> String {
+    let date_time = DateTime::from_timestamp_millis(millis).unwrap();
+    date_time.to_rfc3339_opts(SecondsFormat::Millis, true)
+}
+
+#[Object]
+impl FeetransferQueryRoot {
+    async fn feetransfers<'ctx>(
+        &self,
+        ctx: &Context<'ctx>,
+        query: Option<FeetransferQueryInput>,
+        sort_by: Option<FeetransferSortByInput>,
+        limit: Option<usize>,
+    ) -> Result<Option<Vec<FeetransferWithMeta>>> {
+        let db = ctx
+            .data::<Arc<IndexerStore>>()
+            .expect("db to be in context");
+        let limit = limit.unwrap_or(100);
+
+        let has_state_hash = query.as_ref().map_or(false, |q| q.state_hash.is_some());
+
+        if has_state_hash {
+            let state_hash = query
+                .as_ref()
+                .and_then(|q| q.state_hash.as_ref())
+                .map_or(MAINNET_GENESIS_HASH, |s| s);
+            let state_hash = BlockHash::from(state_hash);
+            return Ok(get_fee_transfers_for_state_hash(
+                db,
+                &state_hash,
+                sort_by,
+                limit,
+            ));
+        }
+        get_fee_transfers(db, query, sort_by, limit)
+    }
+}
+
+fn get_fee_transfers(
+    db: &Arc<IndexerStore>,
+    query: Option<FeetransferQueryInput>,
+    sort_by: Option<FeetransferSortByInput>,
+    limit: usize,
+) -> Result<Option<Vec<FeetransferWithMeta>>> {
+    let mut fee_transfers: Vec<FeetransferWithMeta> = Vec::new();
+    let mode: speedb::IteratorMode = match sort_by {
+        Some(FeetransferSortByInput::BlockHeightAsc) => speedb::IteratorMode::Start,
+        Some(FeetransferSortByInput::BlockHeightDesc) => speedb::IteratorMode::End,
+        None => speedb::IteratorMode::End,
+    };
+    for entry in db.get_internal_commands_interator(mode) {
+        let (_, value) = entry?;
+        let internal_command = serde_json::from_slice::<InternalCommandWithData>(&value)?;
+        let ft = Feetransfer::from(internal_command);
+        let state_hash = ft.state_hash.clone();
+        let canonical = db
+            .get_block_canonicity(&state_hash.into())?
+            .map(|status| matches!(status, Canonicity::Canonical))
+            .unwrap_or(false);
+
+        let should_filter = query
+            .clone()
+            .and_then(|q| q.canonical)
+            .map(|canonicity_filter| canonicity_filter != canonical)
+            .unwrap_or(false);
+
+        if should_filter {
+            continue;
+        }
+
+        fee_transfers.push(FeetransferWithMeta {
+            canonical,
+            feetransfer: ft,
+            block: None,
+        });
+        if fee_transfers.len() >= limit {
+            break;
+        }
```

Comment:
Is this correct? Because the data is already in the correctly sorted order?

Reply:
Yes, I return them based on global_slot_since_genesis order and that's the only sort I support.